### DISCUSSION TITLE
Channeling Live Output Signals into an IBlackBox Input Signal Array

### DISCUSSION THREAD

**Discussion by DatGeekUKnow** [colgreen/sharpneat#45] [Q&A] [answered]
Is there any possible way to enforce a slight feedback loop like this?  Let me explain my application:
I'm creating a Coevolution experiment to learn how to play a board game.
Each turn of the board game has 3 phases; meaning 3 decisions necessary from the Coevolution Neat Player.
The second and third decisions depend widely on the decisions that came before it.  

This makes statically activating the network once given a list of static inputs isn't possible. So the outputs for the first two nodes would be fed into the input nodes 15-17 or something

**Comment by colgreen** [comment, marked as answer]
Yes, it's quite normal to activate the black box more than once within the scope of a single fitness evaluation. The pattern you describe sounds fine to me, i.e., taking outputs, applying them to the input signal array, reactivating the black box, and repeated multiple times.

**Reply by DatGeekUKnow** [reply]
Thank you so much for your prompt response! I'm using this library for part of my senior capstone project, and I wanted to thank you for the wonderful resource!
